boot/makebootable.go

```diff
@@ -379,22 +386,9 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
 		}
 	}
 
-	// LAST step: update recovery bootloader environment to indicate that we
-	// transition to run mode now
-	opts = &bootloader.Options{
-		// let the bootloader know we will be touching the recovery
-		// partition
-		Role: bootloader.RoleRecovery,
-	}
-	bl, err = bootloader.Find(InitramfsUbuntuSeedDir, opts)
-	if err != nil {
-		return fmt.Errorf("internal error: cannot find recovery system bootloader: %v", err)
-	}
-	blVars = map[string]string{
-		"snapd_recovery_mode": "run",
-	}
-	if err := bl.SetBootVars(blVars); err != nil {
-		return fmt.Errorf("cannot set recovery environment: %v", err)
-	}
+	// Note that we do not update the recovery bootloader env to actually
+	// transition to run mode here, that is left to the caller via something
+	// like boot.EnsureNextBootToRunMode()
```

Comment:
this kind of comment should also go in the doc comment for the function itself I think

Reply:
moved this too (forgot initially, so I force pushed to add it to one of my other comments)